hbase10/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -208,14 +210,11 @@ public void cleanup() throws DBException {
       long en = System.nanoTime();
       final String type = clientSideBuffering ? "UPDATE" : "CLEANUP";
       measurements.measure(type, (int) ((en - st) / 1000));
-      threadCount.decrementAndGet();
-      if (threadCount.get() <= 0) {
-        // Means we are done so ok to shut down the Connection.
-        synchronized (CONNECTION_LOCK) {
-          if (connection != null) {
-            connection.close();
-            connection = null;
-          }
+      synchronized (THREAD_COUNT) {
+        int threadCount = THREAD_COUNT.decrementAndGet();
+        if (threadCount <= 0 && connection != null) {
+          connection.close();
+          connection = null;
```

Comment:
This synchronized block should not be needed. The closing of the connection is protected by the decrement of the atomic integer.

Reply:
Good point, yanked the sync.